src/components/BookStatus/BookRecordsUsers/UserRecordsPopper/index.tsx

```diff
@@ -0,0 +1,82 @@
+/* eslint-disable no-unused-vars */
+import { Paper, Popper, Typography } from "@mui/material";
+import { forwardRef, Ref, useImperativeHandle, useState } from "react";
+import { DetailedRecord } from "../../../../store/books/details/Types";
+import BookStatusProperties from "../../../../store/books/status/BookStatusProperties";
+import { BookStatus } from "../../../../store/books/status/Types";
+
+type Props = {
+  records?: DetailedRecord[];
+};
+
+export interface UserRecordsPopperRefObject {
+  showPopper?: (event: any) => void;
+  handleLeave?: () => void;
+}
+
+const UserRecordsPopper = forwardRef(
+  ({ records }: Props, ref: Ref<UserRecordsPopperRefObject>) => {
+    const [anchorEl, setAnchorEl] = useState(null);
+    const open = Boolean(anchorEl);
+
+    const showPopper = (event: any) => {
+      setAnchorEl(event.currentTarget);
+    };
+
+    const handleLeave = () => {
+      setAnchorEl(null);
+    };
+
+    useImperativeHandle(ref, () => ({ showPopper, handleLeave }));
+
+    const getBookStatusProperties = (
+      status: BookStatus | null,
+      userFullName: string | null = ""
+    ) => {
+      return new BookStatusProperties(status, userFullName);
+    };
```

Comment:
We can move this function to BookStatusProperties file and export it from there, so we don't have to define it 3 times.